### PR TITLE
[suggest] Badge suggestions show incorrect license

### PR DESCRIPTION
**Pull request by platan**
This PR tries to resolve https://github.com/badges/shields/issues/821.

### CODE REVIEW

File: service-tests/suggest.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+const Joi = require('joi');
+const ServiceTester = require('./runner/service-tester');
+
+const t = new ServiceTester({ id: 'suggest', title: 'suggest', pathPrefix: '/$suggest' });
+module.exports = t;
+
+
+t.create('issues')
+  .get('/v1?url=' + encodeURIComponent('https://github.com/atom/atom'))
+  .addHeader('origin', 'https://shields.io')
```
Comment (platan) [inline service-tests/suggest.js]: https://github.com/badges/shields/blob/master/lib/suggest.js#L15 requires this header value.

**Comment by platan** [comment]
CI build failed https://travis-ci.org/badges/shields/jobs/286720279. Do you know what could be the cause?

**Comment by paulmelnikow** [comment]
Thanks for this!

I alerted Thaddée earlier today that one of the three Shields production servers appears to be down. That's the cause of the build failure. I restarted the build.

File: lib/github-auth.js [hunk 2]
```diff
@@ -214,14 +214,15 @@ function rmGithubToken(token) {
...
+function githubRequest(request, url, query, cb, customHeaders = {}) {
   query = query || {};
   // A special User-Agent is required:
   // http://developer.github.com/v3/#user-agent-required
-  const headers = {
+  const defaultHeaders = {
     'User-Agent': 'Shields.io',
     'Accept': 'application/vnd.github.v3+json',
   };
+  const headers = Object.assign({}, defaultHeaders, customHeaders);
```
Comment (paulmelnikow) [inline lib/github-auth.js]: Are these changes necessary?

File: lib/suggest.js [hunk 3]
```diff
@@ -89,169 +91,33 @@ function githubStars (user, repo) {
...
-              } catch(e) { reject(e); }
-            });
-          } catch(e) { reject(e); }
-        });
-      } catch(e) { reject(e); }
-    });
+      } catch(e) {
+        resolve(null);
+      }
+    }, { 'Accept' : 'application/vnd.github.drax-preview+json' })
```
Comment (paulmelnikow) [inline lib/suggest.js]: So the idea is to suggest the license badge, but only if the badge will show something?

Assuming the badge shows something like "unknown" if the license isn't specified / detectable, I wonder if it's worth suggesting that badge anyway? It could trigger the developer to correctly indicate the license in their repo.

File: lib/suggest.js [hunk 4]
```diff
@@ -89,169 +91,33 @@ function githubStars (user, repo) {
...
-  return ('https://img.shields.io/badge/'
-    + escapeField(badgeData.text[0])
-    + '-' + escapeField(badgeData.text[1])
-    + '-' + badgeData.colorscheme + '.svg');
-}
-
-function escapeField (s) {
-  return encodeURIComponent(s.replace(/-/g, '--').replace(/_/g, '__'));
-}
-
```
Comment (paulmelnikow) [inline lib/suggest.js]: 🍭

File: service-tests/suggest.js (hunk 1 above)
Comment (paulmelnikow) [reply service-tests/suggest.js]: Worth mentioning that in a comment? ^^

File: service-tests/suggest.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  });
+
+t.create('forks')
+  .get('/v1?url=' + encodeURIComponent('https://github.com/atom/atom'))
+  .addHeader('origin', 'https://shields.io')
+  .expectJSON('badges.?', {
+    name: 'GitHub forks',
+    link: 'https://github.com/atom/atom/network',
+    badge: 'https://img.shields.io/github/forks/atom/atom.svg'
+  });
```
Comment (paulmelnikow) [inline service-tests/suggest.js]: Would it work to combine all of these into a single `t.create()`?

```js
t.create('issues')
  .get('/v1?url=' + encodeURIComponent('https://github.com/atom/atom'))
  .addHeader('origin', 'https://shields.io')
  .expectJson(...)
  .expectJson(...)
  .expectJson(...)
```

File: service-tests/suggest.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    link: 'https://github.com/atom/atom/stargazers',
+    badge: 'https://img.shields.io/github/stars/atom/atom.svg'
+  });
+
+t.create('twitter')
+  .get('/v1?url=' + encodeURIComponent('https://github.com/atom/atom'))
+  .addHeader('origin', 'https://shields.io')
+  .expectJSON('badges.?', {
+    name: 'Twitter',
+    link: 'https://twitter.com/intent/tweet?text=Wow:&url=%5Bobject%20Object%5D',
```
Comment (paulmelnikow) [inline service-tests/suggest.js]: `object Object` looks like a bug.

File: lib/github-auth.js (hunk 2 above)
Comment (platan) [reply lib/github-auth.js]: I wanted to pass custom header to this method. Licenses API requires this header https://developer.github.com/v3/licenses/. Without it I wouldn't get license file url.

File: service-tests/suggest.js (hunk 6 above)
Comment (platan) [reply service-tests/suggest.js]: I've fixed it. Thanks!

File: service-tests/suggest.js (hunk 5 above)
Comment (platan) [reply service-tests/suggest.js]: I've chained issues, forks, stars and twitter.

File: lib/suggest.js (hunk 3 above)
Comment (platan) [reply lib/suggest.js]: I need to access License API to URL to license file. But you're right, we can always suggest license badge. I like this idea. I will fix it.

File: lib/github-auth.js (hunk 2 above)
Comment (paulmelnikow) [reply lib/github-auth.js]: You don't seem to be doing that though. What am I missing?

Comment (platan) [reply lib/github-auth.js]: In `lib/suggest.js`, in function `githubLicense` I'm invoking `githubAuth.request` with 5 parameters and the last one (after callback) is Accept header.

Comment (platan) [reply lib/github-auth.js]: But maybe it is not required. Give me a moment...

Comment (paulmelnikow) [reply lib/github-auth.js]: Ahhh I see, it's after the callback.

That's super confusing and too easy to miss. I'd probably suggest moving that parameter before the callback, and updating all the other calls to pass an empty object. That's consistent with the way we handle `query`.

Alternatively you could do something clever like `request`, which detects whether each parameter is an option or the callback. I think this is more trouble than it's worth, especially because I'm hoping we rewrite all this stuff using promises and async/await pretty soon.

Comment (platan) [reply lib/github-auth.js]: I know it's confusing, but I didn't want to change method parameters order (a lot of code/people using it). 
In the meantime, I removed this custom headers. With default media type in the Accept header API returns right data. 
Sorry for this changes.

Comment (paulmelnikow) [reply lib/github-auth.js]: Even better!

**Comment by paulmelnikow** [comment]
Thanks!

**Comment by platan** [comment]
Than you for your time and detailed review!

**Comment by paulmelnikow** [comment]
It was my pleasure!
